### PR TITLE
add json file that contains build information

### PR DESCRIPTION
**Pull request by neophob**
Idea:
- browser: use build.json in webpack build step
- node: use require to load the information from buid.json

### CODE REVIEW

**Comment by neophob** [comment]
vpweb pr: https://github.com/freezy/vpweb/pull/34

**Comment by neophob** [comment]
Whats missing is the watch step need to rebuild the build.json file

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
 		"build:cjs": "tsc --module commonjs --outDir ./dist/cjs",
 		"build:declarations": "tsc --module es2015 --outDir ./dist/esm --declaration true --allowJs false",
 		"build:watch": "npm run compile && npm run build:cp && tsc --watch --module es2015 --outDir ./dist/esm",
-		"build:cp": "cp -r res dist/esm && cp -r res dist/cjs",
+		"build:cp": "cp -r res dist/esm && cp -r res dist/cjs && node build/addBuildInformation.js",
```
Comment (freezy) [inline package.json]: Now it doesn't only copy anymore, how about calling it `build:post`?

**Comment by freezy** [comment]
Hmm after checking the vpweb PR, just reading the version from `build.json` instead of an API is good enough I think, and avoids having an ignored `build.json` at the project root.

File: package.json [hunk 2]
```diff
@@ -32,12 +32,13 @@
 		"./dist/esm/lib/refs.node.js": "./dist/esm/lib/refs.browser.js"
 	},
 	"scripts": {
-		"build": "rimraf dist && npm run compile && npm run build:esm && npm run build:cjs && npm run build:declarations && npm run build:cp",
+		"build": "rimraf dist && npm run compile && npm run build:esm && npm run build:cjs && npm run build:declarations && npm run build:cp && npm run build:post",
 		"build:esm": "tsc --module es2015 --outDir ./dist/esm",
 		"build:cjs": "tsc --module commonjs --outDir ./dist/cjs",
 		"build:declarations": "tsc --module es2015 --outDir ./dist/esm --declaration true --allowJs false",
 		"build:watch": "npm run compile && npm run build:cp && tsc --watch --module es2015 --outDir ./dist/esm",
```
Comment (freezy) [inline package.json]: I think you gotta do a `build:post` here too, otherwise `:watch` won't generate a `build.json`.

Comment (neophob) [reply package.json]: ill handle that in another pr - there are also vpweb changes needed. using tsc-watch

Comment (freezy) [reply package.json]: fair enough!

**Pull request merged**